ros_packages/pibsim_webots/pibsim_webots/pib_driver.py

```diff
@@ -1,148 +1,121 @@
+        "wrist_right": ["wrist_right"],
+    }
+
+    def __initialize_devices(self) -> dict:
+        """Initialize all webot robot devices associated with the pib robot names"""
+        devices = {}
+
+        for pib_motor_name, proto_device_names in self.MOTOR_TO_DEVICES_MAP.items():
+            devices[pib_motor_name] = self.find_devices_in_robot(proto_device_names)
+        return devices
+
+    def __trajectory_callback(self, joint_trajectory: JointTrajectory):
+        """Sets the positions of all devices in a received joint trajectory message."""
+        motor_name_position_pairs = self.extract_motor_name_position_pairs(
+            joint_trajectory
         )
 
-    def convert_cerebra(self, position):
+        for (
+            motor_name,
+            device_target_position,
+        ) in motor_name_position_pairs:
+            self.set_device_positions(
+                motor_name, self.convert_position(device_target_position)
+            )
+
+    def find_devices_in_robot(self, device_names: list) -> list:
+        """Retrieves devices from the robot based on a list of device names"""
+        device_list = []
+
+        for device_name in device_names:
+            device = self.__robot.getDevice(device_name)
+            if device is not None:
+                device_list.append(device)
+            else:
+                self.__node.get_logger().error(
+                    f"Failed to retrieve device: {device_name}"
+                )
+        return device_list
+
+    def convert_position(self, position: float) -> float:
+        """Converts degree to radian and adjust for cerebras offset (multiplying degree by 100)."""
         return math.radians(position / 100.0)
 
-    def create_device(self, name):
-        rotIndex = name.find("rota")
-        if rotIndex != -1:
-            device = name[: (rotIndex - 1)]
-        else:
-            device = name
-        self.__devices[name] = [
-            self.__robot.getDevice(device),
-        ]
-        self.__node.get_logger().info("created device: " + device + "   name: " + name)
-
-    # position interface, only the first point in a trajectory is considered
-    def set_target_positions(self):
-        for index in range(len(self.__target_trajectory.joint_names)):
-            name = self.__target_trajectory.joint_names[index]
-            if not (name in self.__devices):
-                self.create_device(name)
-
-            position = self.convert_cerebra(
-                self.__target_trajectory.points[0].positions[index]
+    def extract_motor_name_position_pairs(
+        self, jt: JointTrajectory
+    ) -> Iterable[Tuple[str, int]]:
+        """unpacks a jt-message into an iterable of motor_name-position-pairs"""
+        motor_names = jt.joint_names
+        points: list[JointTrajectoryPoint] = jt.points
+        positions = (point.positions[0] for point in points)
+
+        motor_name_position_pairs = zip(motor_names, positions)
```

Comment:
Do `motor_names` and `positions` always have the same length?

Reply:
If the JointTrajectory Message is properly formed and adheres to the schema defined by ROS then yes. 
Each entry in the names array of the JT-Message should exactly correlate to one JointTrajectoryPoint and thus a position. 

Although it is possible to manually create an invalid JT-Message where there are more names than positions or vice versa.
Should this case be handled?

Reply:
If ROS doesn't reject the invalid JT-Messages I would handle it just in case with a simple if check with early return and error logging.

Reply:
I quickly tested this and it appears that ROS does not validate weather the number of jointNames matches the number of jointTrajectoryPoints. Meaning wrong jt-messages can be published onto a corresponding topic.

I implemented a simple if-statement as suggested.